src/Microsoft.ML.Transforms/Text/WordEmbeddingsTransform.cs

```diff
@@ -121,46 +128,127 @@ public bool GetWordVector(ref DvText word, float[] wordVector)
 
         /// <summary>
-        /// Public constructor corresponding to <see cref="SignatureDataTransform"/>.
+        /// Instantiates <see cref="WordEmbeddingsTransform"/> using the pretrained word embedding model specified by <paramref name="modelKind"/>.
+        /// </summary>
+        /// <param name="env">Host Environment.</param>
+        /// <param name="inputColumn">Name of the input column.</param>
+        /// <param name="outputColumn">Name of the output column.</param>
+        /// <param name="modelKind">The pretrained word embedding model.</param>
+        public WordEmbeddingsTransform(IHostEnvironment env, string inputColumn, string outputColumn,
+           PretrainedModelKind modelKind = PretrainedModelKind.Sswe)
+           : this(env, modelKind, new ColumnInfo(inputColumn, outputColumn))
+        {
+        }
+
+        /// <summary>
+        /// Instantiates <see cref="WordEmbeddingsTransform"/> using the custom word embedding model by loading it from the file specified by the <paramref name="customLookupTable"/>.
         /// </summary>
-        public WordEmbeddingsTransform(IHostEnvironment env, Arguments args, IDataView input)
-                : base(env, RegistrationName, Contracts.CheckRef(args, nameof(args)).Column,
-                     input, TestIsTextVector)
+        /// <param name="env">Host Environment.</param>
+        /// <param name="inputColumn">Name of the input column.</param>
+        /// <param name="outputColumn">Name of the output column.</param>
+        /// <param name="customLookupTable">Filename for custom word embedding model.</param>
+        public WordEmbeddingsTransform(IHostEnvironment env, string inputColumn, string outputColumn, string customLookupTable)
+           : this(env, customLookupTable, new ColumnInfo(inputColumn, outputColumn))
         {
+        }
+
+        /// <summary>
+        /// Instantiates <see cref="WordEmbeddingsTransform"/> using the pretrained word embedding model specified by <paramref name="modelKind"/>.
+        /// </summary>
+        /// <param name="env">Host Environment.</param>
+        /// <param name="modelKind">The pretrained word embedding model.</param>
+        /// <param name="columns">Input/Output columns.</param>
+        public WordEmbeddingsTransform(IHostEnvironment env, PretrainedModelKind? modelKind, params ColumnInfo[] columns)
+            : base(Contracts.CheckRef(env, nameof(env)).Register(RegistrationName), GetColumnPairs(columns))
+        {
+            if (modelKind == null)
+                modelKind = PretrainedModelKind.Sswe;
+            env.CheckUserArg(modelKind.HasValue || Enum.IsDefined(typeof(PretrainedModelKind), modelKind), nameof(modelKind));
+
+            _modelKind = modelKind;
+            _modelFileNameWithPath = EnsureModelFile(env, out _linesToSkip, (PretrainedModelKind)_modelKind);
+            Host.CheckNonWhiteSpace(_modelFileNameWithPath, nameof(_modelFileNameWithPath));
```

Comment:
>Host.CheckNonWhiteSpace(_modelFileNameWithPath, nameof(_modelFileNameWithPath)); [](start = 12, length = 80)

why this is not responsibility of EnsureModelFile?

Reply:
Looked at the code inside EnsureModelFile and found out that if the call is successful it will always have correct FileName otherwise throw error. I am going to remove this check.

---
In reply to: [218253058](https://github.com/dotnet/machinelearning/pull/928#discussion_r218253058) [](ancestors = 218253058)